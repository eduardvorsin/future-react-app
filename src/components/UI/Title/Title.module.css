.title {
	font-family: Vollkorn, serif;
	text-align: center;
	font-weight: 700;
	margin: 0;
}

.title--dark {
	color: var(--dark-color-100);
}

.title--light {
	color: var(--light-color-100);
}

.title--error {
	color: var(--error-color);
}

.title--level-1 {
	font-size: var(--font-size-title-1);
}

.title--level-2 {
	font-size: var(--font-size-title-2);
}

.title--level-3 {
	font-size: var(--font-size-title-3);
}

.title--level-4 {
	font-size: var(--font-size-title-4);
}

.title--level-5 {
	font-size: var(--font-size-title-5);
}

.title--level-6 {
	font-size: var(--font-size-title-6);
}