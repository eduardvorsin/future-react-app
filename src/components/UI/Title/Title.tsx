import React from 'react';
import { isValidHeadingLevel } from '../../../helpers/helpers';
import classes from './Title.module.css';
import { Theme } from '../../../contexts/ThemeContext';

type BaseTitleProps = Partial<React.HTMLAttributes<HTMLHeadingElement>>;
type HeadingLevels = 1 | 2 | 3 | 4 | 5 | 6;
type Headings = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

interface TitleProps extends BaseTitleProps {
  children: React.ReactNode,
  level: HeadingLevels,
  component?: Headings,
  variant?: Theme,
  className?: string,
}

const Title: React.FC<TitleProps> = ({
  level,
  children,
  className,
  component = 'h6',
  variant = 'dark',
  ...props
}) => {
  const headingClasses = [
    classes.title,
    className,
    classes[`title--${variant}`],
    isValidHeadingLevel(level) ? classes[`title--level-${level}`] : classes['title--level-6'],
  ].join(' ');

  const Heading = component;

  return (
    <Heading
      className={headingClasses}
      {...props}
    >
      {children}
    </Heading>
  );
};

export default Title;
