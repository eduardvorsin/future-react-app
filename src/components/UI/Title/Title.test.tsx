import React from 'react';
import { render, screen } from '@testing-library/react';
import Title from './Title';

describe('Title tests', () => {
  it('is rendered correctly', () => {
    render(<Title level={4}>test title</Title>);

    expect(screen.getByRole<HTMLHeadingElement>('heading', { name: /test title/i })).toBeInTheDocument();
  });
  it('if the variant prop is equal to light, then the title--light class is added', () => {
    render(
      <Title
        variant='light'
        level={4}
      >
        test title
      </Title>,
    );

    expect(screen.getByRole<HTMLHeadingElement>('heading', { name: /test title/i })).toHaveClass('title--light');
  });
  it('if the level prop is set with the value 3, the title--level-3 class will be added', () => {
    render(
      <Title
        level={3}
      >
        test title
      </Title>,
    );

    expect(screen.getByRole<HTMLHeadingElement>('heading', { name: /test title/i })).toHaveClass('title--level-3');
  });
  it('is a basic snapshot', () => {
    render(
      <Title
        level={6}
        className='test-title'
      >
        test title
      </Title>,
    );

    expect(screen.getByRole<HTMLHeadingElement>('heading', { name: /test title/i })).toMatchSnapshot();
  });
});
