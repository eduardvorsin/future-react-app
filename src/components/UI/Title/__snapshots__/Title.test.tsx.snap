// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title tests is a basic snapshot 1`] = `
<h6
  class="title test-title title--dark title--level-6"
>
  test title
</h6>
`;
